controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -463,7 +464,10 @@ func (r *BareMetalHostReconciler) actionRegistering(prov provisioner.Provisioner
 	registeredNewCreds := !info.host.Status.GoodCredentials.Match(*info.bmcCredsSecret)
 	info.host.UpdateGoodCredentials(*info.bmcCredsSecret)
 	info.log.Info("clearing previous error message")
-	clearError(info.host)
+
+	if info.host.Status.ErrorType == metal3v1alpha1.RegistrationError || registeredNewCreds {
+		clearError(info.host)
+	}
```

Comment:
Suggestion: if we're going to fix the issue here you then if you want tests from #757 can be reused (and I'm going to update it accordingly just to fix the log traces)

Reply:
Ok makes sense - I pulled in your commit over without the logging changes, although the unit test is failing so I guess good call bringing the test over. Not sure what I broke, though.

Reply:
I should have fixed them in #757 (due the latest rebase)